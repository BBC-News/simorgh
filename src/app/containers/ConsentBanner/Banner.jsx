import React from 'react';
import { func, string, bool } from 'prop-types';
import * as AmpHelpers from 'react-amphtml/helpers';
import { ServiceContextConsumer } from '../../contexts/ServiceContext';
import {
  ConsentBanner,
  ConsentBannerText,
} from '../../components/ConsentBanner';

const Accept = (message, onClick) => (
  <AmpHelpers.Action events={onClick}>
    {props => (
      <button {...props} type="button">
        {message}
      </button>
    )}
  </AmpHelpers.Action>
);

const Reject = (message, href, onClick) => (
  <AmpHelpers.Action events={onClick}>
    {props => (
      <a {...props} href={href}>
        {message}
      </a>
    )}
  </AmpHelpers.Action>
);

const Text = ({ first, linkText, linkUrl, last }) => {
  const Link = linkUrl && linkText ? <a href={linkUrl}>{linkText}</a> : null;

  return (
    <ConsentBannerText>
      {first}
      {Link}
      {last}
    </ConsentBannerText>
  );
};

Text.propTypes = {
  first: string.isRequired,
  linkText: string.isRequired,
  linkUrl: string.isRequired,
  last: string.isRequired,
};

const ConsentBannerContainer = ({
  type,
  onReject,
  onAccept,
  promptId,
  hidden,
}) => (
  <ServiceContextConsumer>
    {({ translations }) => {
      const messaging = translations.consentBanner[type];

      return (
        <ConsentBanner
          id={promptId}
          title={messaging.title}
          text={Text(messaging.description)}
          accept={Accept(messaging.accept, onAccept)}
          reject={Reject(messaging.reject, messaging.rejectUrl, onReject)}
          hidden={hidden}
        />
      );
    }}
  </ServiceContextConsumer>
);

ConsentBannerContainer.propTypes = {
  type: string.isRequired,
  onReject: func.isRequired,
  onAccept: func.isRequired,
  promptId: string,
  hidden: bool,
};

ConsentBannerContainer.defaultProps = {
  promptId: null,
  hidden: null,
};

export default ConsentBannerContainer;
