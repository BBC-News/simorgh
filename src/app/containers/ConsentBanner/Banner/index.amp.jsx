import React, { useContext } from 'react';
import { bool, string } from 'prop-types';
import { ConsentBanner } from '@bbc/psammead-consent-banner';
import { ServiceContext } from '../../../contexts/ServiceContext';
import BannerText from './Text';

const Accept = (message, onClick) => (
  <button type="button" on={onClick}>
    {message}
  </button>
);

const Reject = (message, href, onClick) => (
  <a href={href} on={onClick}>
    {message}
  </a>
);

const AmpConsentBannerContainer = ({
  type,
  acceptAction,
  rejectAction,
  promptId,
  hidden,
}) => {
  const { translations, script } = useContext(ServiceContext);
  const consentBannerConfig = translations.consentBanner[type];

  return (
    <ConsentBanner
      id={promptId}
      title={consentBannerConfig.title}
      text={BannerText(consentBannerConfig.description)}
      accept={Accept(consentBannerConfig.accept, acceptAction)}
      reject={Reject(
        consentBannerConfig.reject,
        consentBannerConfig.rejectUrl,
        rejectAction,
      )}
      hidden={hidden}
      script={script}
    />
  );
};

AmpConsentBannerContainer.propTypes = {
  type: string.isRequired,
  acceptAction: string.isRequired,
  rejectAction: string.isRequired,
  promptId: string.isRequired,
  hidden: bool,
};

AmpConsentBannerContainer.defaultProps = {
  hidden: null,
};

export default AmpConsentBannerContainer;
