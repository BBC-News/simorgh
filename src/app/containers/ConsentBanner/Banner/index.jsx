import React from 'react';
import { func, string } from 'prop-types';
import { ServiceContextConsumer } from '../../../contexts/ServiceContext';
import {
  ConsentBanner,
  ConsentBannerButton,
  ConsentBannerLink,
} from '../../../components/ConsentBanner';
import BannerText from './Text';

const Accept = (message, onClick) => (
  <ConsentBannerButton onClick={onClick}>{message}</ConsentBannerButton>
);

const Reject = (message, href, onClick) => (
  <ConsentBannerLink href={href} onClick={onClick}>
    {message}
  </ConsentBannerLink>
);

const ConsentBannerContainer = ({ type, onReject, onAccept }) => (
  <ServiceContextConsumer>
    {({ translations }) => {
      const consentBannerConfig = translations.consentBanner[type];

      return (
        <ConsentBanner
          title={consentBannerConfig.title}
          text={BannerText(consentBannerConfig.description)}
          accept={Accept(consentBannerConfig.accept, onAccept)}
          reject={Reject(
            consentBannerConfig.reject,
            consentBannerConfig.rejectUrl,
            onReject,
          )}
        />
      );
    }}
  </ServiceContextConsumer>
);

ConsentBannerContainer.propTypes = {
  type: string.isRequired,
  onReject: func.isRequired,
  onAccept: func.isRequired,
};

export default ConsentBannerContainer;
