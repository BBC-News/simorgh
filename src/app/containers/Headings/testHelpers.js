const blocksSingleFragment = (text, attributes = []) => [
  {
    id: '7987213',
    type: 'text',
    model: {
      blocks: [
        {
          id: '6897239',
          type: 'paragraph',
          model: {
            text,
            blocks: [
              {
                id: '4209872',
                type: 'fragment',
                model: {
                  text,
                  attributes,
                },
              },
            ],
          },
        },
      ],
    },
  },
];

export default blocksSingleFragment;
