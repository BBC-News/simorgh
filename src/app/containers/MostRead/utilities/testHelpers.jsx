import React from 'react';
import MostReadContainer from '..';
import { RequestContextProvider } from '#contexts/RequestContext';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import { ToggleContext } from '#contexts/ToggleContext';

export const setStalePromoTimestamp = mostReadData => {
  const oldTimestamp = 864691200; // 27/05/1997
  const updatedMostReadData = mostReadData;

  // set first promo to have an old timestamp
  updatedMostReadData.records[0].promo.timestamp = oldTimestamp;

  return updatedMostReadData;
};

export const setFreshPromoTimestamp = mostReadData => {
  const freshDate = new Date();
  const updatedMostReadData = mostReadData;

  // Updates first 10 promos to have a fresh date
  for (let i = 0; i < 10; i += 1) {
    updatedMostReadData.records[i].promo.timestamp = freshDate.getTime();
  }

  return updatedMostReadData;
};

export const setStaleLastRecordTimeStamp = mostReadData => {
  const updatedMostReadData = mostReadData;
  updatedMostReadData.lastRecordTimeStamp = '2019-11-06T16:28:00Z';

  return updatedMostReadData;
};

const getToggleState = enabled => ({
  local: { mostRead: { enabled } },
  test: { mostRead: { enabled } },
});

/* eslint-disable react/prop-types */
export const MostReadWithContext = ({
  isAmp = false,
  service,
  variant = null,
  mostReadToggle = false,
}) => (
  <ToggleContext.Provider
    value={{ toggleState: getToggleState(mostReadToggle) }}
  >
    <RequestContextProvider
      bbcOrigin={`http://localhost:7080/${service}`}
      isAmp={isAmp}
      pageType="frontPage"
      service={service}
      statusCode={200}
      pathname={`/${service}`}
      variant={variant}
    >
      <ServiceContextProvider service={service} variant={variant}>
        <MostReadContainer />
      </ServiceContextProvider>
    </RequestContextProvider>
  </ToggleContext.Provider>
);
/* eslint-enable react/prop-types */
