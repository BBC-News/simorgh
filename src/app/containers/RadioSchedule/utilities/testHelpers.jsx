import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import RadioScheduleContainer from '..';
import { RequestContextProvider } from '#contexts/RequestContext';
import { ServiceContextProvider } from '#contexts/ServiceContext';
import { ToggleContext } from '#contexts/ToggleContext';

const getToggleState = enabled => ({
  radioSchedule: { enabled },
});

/* eslint-disable react/prop-types */
const RadioSchedulesWithContext = ({
  service,
  radioScheduleToggle = false,
}) => (
  <ToggleContext.Provider
    value={{
      toggleState: getToggleState(radioScheduleToggle),
      toggleDispatch: jest.fn(),
    }}
  >
    <RequestContextProvider
      isAmp={false}
      pageType="frontPage"
      service={service}
      pathname={`/${service}`}
      timeOnServer={Date.now()}
    >
      <ServiceContextProvider service={service}>
        <BrowserRouter>
          <RadioScheduleContainer />
        </BrowserRouter>
      </ServiceContextProvider>
    </RequestContextProvider>
  </ToggleContext.Provider>
);
/* eslint-enable react/prop-types */

export default RadioSchedulesWithContext;
