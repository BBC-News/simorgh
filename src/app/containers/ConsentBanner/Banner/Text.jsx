import React, { useContext } from 'react';
import { string, shape } from 'prop-types';
import { RequestContext } from '../../../contexts/RequestContext';
import { ConsentBannerText } from '../../../components/ConsentBanner';

const BannerText = ({ uk, international }) => {
  const { isUK } = useContext(RequestContext);

  const { first, linkText, linkUrl, last } = isUK ? uk : international;

  const Link = linkUrl && linkText ? <a href={linkUrl}>{linkText}</a> : null;

  return (
    <ConsentBannerText>
      {first}
      {Link}
      {last}
    </ConsentBannerText>
  );
};

const messagingProps = {
  first: string.isRequired,
  linkText: string,
  linkUrl: string,
  last: string,
};

BannerText.propTypes = {
  uk: shape(messagingProps).isRequired,
  international: shape(messagingProps).isRequired,
};

export default BannerText;
