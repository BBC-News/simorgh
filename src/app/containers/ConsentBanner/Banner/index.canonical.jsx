import React, { useContext } from 'react';
import { func, string } from 'prop-types';
import { ServiceContext } from '../../../contexts/ServiceContext';
import { ConsentBanner } from '../../../components/ConsentBanner';
import BannerText from './Text';

const Accept = (message, onClick) => (
  <button onClick={onClick} type="button">
    {message}
  </button>
);

const Reject = (message, href, onClick) => (
  <a href={href} onClick={onClick}>
    {message}
  </a>
);

const CanonicalConsentBannerContainer = ({ type, onReject, onAccept }) => {
  const { translations } = useContext(ServiceContext);
  const consentBannerConfig = translations.consentBanner[type];
  return (
    <ConsentBanner
      title={consentBannerConfig.title}
      text={BannerText(consentBannerConfig.description)}
      accept={Accept(consentBannerConfig.accept, onAccept)}
      reject={Reject(
        consentBannerConfig.reject,
        consentBannerConfig.rejectUrl,
        onReject,
      )}
    />
  );
};

CanonicalConsentBannerContainer.propTypes = {
  type: string.isRequired,
  onReject: func.isRequired,
  onAccept: func.isRequired,
};

export default CanonicalConsentBannerContainer;
