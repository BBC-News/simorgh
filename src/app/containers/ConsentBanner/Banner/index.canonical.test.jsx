import React from 'react';
import { shouldMatchSnapshot } from '../../../helpers/tests/testHelpers';
import { ServiceContextProvider } from '../../../contexts/ServiceContext';
import Banner from './index.canonical';

describe('Canonical Consent Banner Container - Banner', () => {
  shouldMatchSnapshot(
    'should correctly render privacy banner',
    <ServiceContextProvider service="news">
      <Banner type="privacy" onAccept={() => {}} onReject={() => {}} />
    </ServiceContextProvider>,
  );

  shouldMatchSnapshot(
    'should correctly render cookie banner',
    <ServiceContextProvider service="news">
      <Banner type="cookie" onAccept={() => {}} onReject={() => {}} />
    </ServiceContextProvider>,
  );
});
