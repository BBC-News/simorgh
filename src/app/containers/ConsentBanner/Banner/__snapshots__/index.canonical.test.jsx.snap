// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Canonical Consent Banner Container - Banner should correctly render cookie banner 1`] = `
.c0 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  background-color: #323232;
  padding: 1rem 0.5rem;
}

.c1 {
  max-width: 80rem;
  margin: 0 auto;
}

.c1::after {
  content: '\\0020';
  display: block;
  height: 0;
  clear: both;
  overflow: hidden;
  visibility: hidden;
}

.c2 {
  font-size: 1.125rem;
  line-height: 1.375rem;
  color: #FFFFFF;
  font-weight: 700;
  padding: 0;
  margin: 0;
}

.c4 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  color: #F6A21D;
  font-weight: 600;
  padding: 0;
  margin: 0;
  list-style-type: none;
}

.c4 li + li {
  padding-top: 0.5rem;
}

.c3 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  color: #BEBEBE;
}

.c3 a {
  color: #F6A21D;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3 a:focus,
.c3 a:hover {
  color: #FFFFFF;
}

.c5 button {
  font-size: 1.125rem;
  line-height: 1.375rem;
  color: #F6A21D;
  font-weight: 700;
  background: none;
  border: none;
  padding: 0;
  margin: 0;
  cursor: pointer;
}

.c5 button:focus,
.c5 button:hover {
  color: #FFFFFF;
}

.c5 a {
  color: #F6A21D;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5 a:focus,
.c5 a:hover {
  color: #FFFFFF;
}

@media (min-width:25rem) {
  .c0 {
    padding: 1rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    float: left;
    margin-right: 3.5%;
    width: 22%;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    width: 18%;
    float: right;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c3 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    margin: 0;
    float: left;
    width: 53%;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c5 button {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width:37.5rem) {
  .c5 button {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

<div
  className="c0"
  dir="ltr"
  hidden={null}
  id={null}
>
  <div
    className="c1"
    dir="ltr"
  >
    <h2
      className="c2"
      dir="ltr"
    >
      Let us know you agree to cookies
    </h2>
    <p
      className="c3"
      dir="ltr"
    >
      We and our partners use technologies, such as 
      <a
        href="https://www.bbc.co.uk/usingthebbc/cookies/what-do-i-need-to-know-about-cookies/"
      >
        cookies
      </a>
      , and collect browsing data to give you the best online experience and to personalise the content and advertising shown to you. Please let us know if you agree.
    </p>
    <ul
      className="c4"
      dir="ltr"
    >
      <li
        className="c5"
        dir="ltr"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          Yes, I agree
        </button>
      </li>
      <li
        className="c5"
        dir="ltr"
      >
        <a
          href="https://www.bbc.co.uk/usingthebbc/cookies/how-can-i-change-my-bbc-cookie-settings/"
          onClick={[Function]}
        >
          No, take me to settings
        </a>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Canonical Consent Banner Container - Banner should correctly render privacy banner 1`] = `
.c0 {
  font-family: ReithSans,Helvetica,Arial,sans-serif;
  font-weight: 400;
  font-style: normal;
  background-color: #323232;
  padding: 1rem 0.5rem;
}

.c1 {
  max-width: 80rem;
  margin: 0 auto;
}

.c1::after {
  content: '\\0020';
  display: block;
  height: 0;
  clear: both;
  overflow: hidden;
  visibility: hidden;
}

.c2 {
  font-size: 1.125rem;
  line-height: 1.375rem;
  color: #FFFFFF;
  font-weight: 700;
  padding: 0;
  margin: 0;
}

.c4 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  color: #F6A21D;
  font-weight: 600;
  padding: 0;
  margin: 0;
  list-style-type: none;
}

.c4 li + li {
  padding-top: 0.5rem;
}

.c3 {
  font-size: 0.9375rem;
  line-height: 1.125rem;
  color: #BEBEBE;
}

.c3 a {
  color: #F6A21D;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3 a:focus,
.c3 a:hover {
  color: #FFFFFF;
}

.c5 button {
  font-size: 1.125rem;
  line-height: 1.375rem;
  color: #F6A21D;
  font-weight: 700;
  background: none;
  border: none;
  padding: 0;
  margin: 0;
  cursor: pointer;
}

.c5 button:focus,
.c5 button:hover {
  color: #FFFFFF;
}

.c5 a {
  color: #F6A21D;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c5 a:focus,
.c5 a:hover {
  color: #FFFFFF;
}

@media (min-width:25rem) {
  .c0 {
    padding: 1rem;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c2 {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

@media (min-width:37.5rem) {
  .c2 {
    float: left;
    margin-right: 3.5%;
    width: 22%;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c4 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c4 {
    width: 18%;
    float: right;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c3 {
    font-size: 0.9375rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    font-size: 0.875rem;
    line-height: 1.125rem;
  }
}

@media (min-width:37.5rem) {
  .c3 {
    margin: 0;
    float: left;
    width: 53%;
  }
}

@media (min-width:20rem) and (max-width:37.4375rem) {
  .c5 button {
    font-size: 1.125rem;
    line-height: 1.375rem;
  }
}

@media (min-width:37.5rem) {
  .c5 button {
    font-size: 1.25rem;
    line-height: 1.5rem;
  }
}

<div
  className="c0"
  dir="ltr"
  hidden={null}
  id={null}
>
  <div
    className="c1"
    dir="ltr"
  >
    <h2
      className="c2"
      dir="ltr"
    >
      We've updated our Privacy and Cookies Policy
    </h2>
    <p
      className="c3"
      dir="ltr"
    >
      We've made some important changes to our Privacy and Cookies Policy and we want you to know what this means for you and your data.
    </p>
    <ul
      className="c4"
      dir="ltr"
    >
      <li
        className="c5"
        dir="ltr"
      >
        <button
          onClick={[Function]}
          type="button"
        >
          OK
        </button>
      </li>
      <li
        className="c5"
        dir="ltr"
      >
        <a
          href="https://www.bbc.co.uk/usingthebbc/your-data-matters"
          onClick={[Function]}
        >
          Find out what's changed
        </a>
      </li>
    </ul>
  </div>
</div>
`;
